src/NServiceBus.Testing.Tests/Sagas/ReplyToOriginator.cs

```diff
@@ -0,0 +1,84 @@
+﻿namespace NServiceBus.Testing.Tests.Sagas
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using NUnit.Framework;
+
+    [TestFixture]
+    public class ReplyToOriginator
+    {
+        [Test]
+        public async Task ReplyToOriginatorShouldReplyToInitialOriginator()
+        {
+            const string originatorAddress = "expectedReplyAddress";
+
+            var saga = new TestableSaga<ReplyingSaga, ReplyingSagaData>();
+            // the Originator value is populated by the header value, not the context property
+            await saga.Handle(new StartSagaMessage() { CorrelationProperty = Guid.NewGuid() }, messageHeaders: new Dictionary<string, string>()
+            {
+                {Headers.ReplyToAddress, originatorAddress}
+            });
+            var result = await saga.HandleQueuedMessage();
+
+            var reply = result.Context.RepliedMessages.SingleOrDefault();
+            Assert.NotNull(reply);
+            Assert.AreEqual(originatorAddress, reply.Options.GetDestination());
+            Assert.AreEqual(originatorAddress, reply.Message<ReplyMessage>().OriginatorAddress);
+        }
+
+        [Test]
+        public async Task OriginatorShouldBeSetByDefault()
+        {
+            // ensure the testing API also works without explicitly defining a replyTo header value
+            var saga = new TestableSaga<ReplyingSaga, ReplyingSagaData>();
+
+            await saga.Handle(new StartSagaMessage() { CorrelationProperty = Guid.NewGuid() });
+            var result = await saga.HandleQueuedMessage();
+
+            var reply = result.Context.RepliedMessages.SingleOrDefault();
+            Assert.NotNull(reply);
+            string replyAddress = reply.Options.GetDestination();
+            Assert.NotNull(replyAddress);
+            Assert.AreEqual(replyAddress, reply.Message<ReplyMessage>().OriginatorAddress);
+        }
+
+        class ReplyingSaga : NServiceBus.Saga<ReplyingSagaData>, IAmStartedByMessages<StartSagaMessage>, IHandleMessages<SendReplyMessage>
+        {
+            protected override void ConfigureHowToFindSaga(SagaPropertyMapper<ReplyingSagaData> mapper) => mapper
+                .ConfigureMapping<StartSagaMessage>(m => m.CorrelationProperty)
+                .ToSaga(s => s.CorrelationProperty);
+
+            public Task Handle(StartSagaMessage message, IMessageHandlerContext context)
+            {
+                return context.SendLocal(new SendReplyMessage());
+            }
+
+            public Task Handle(SendReplyMessage message, IMessageHandlerContext context)
+            {
+                return ReplyToOriginator(context, new ReplyMessage { OriginatorAddress = Data.Originator });
```

Comment:
Setting the values in the ReplyMessage is only useful for the asserts, and not required to make ReplyToOriginator work in this case, correct?

Reply:
yes, I've added that to have a simple way to verify the value of `Data.Originator` outside the usage of `ReplyToOriginator`. That property is also publicly accessible.